server/encoder/video_encoder_vulkan_h265.cpp

```diff
@@ -418,6 +418,13 @@ void * wivrn::video_encoder_vulkan_h265::encode_info_next(uint32_t frame_num, si
 
 		reference_lists_info.num_ref_idx_l0_active_minus1 = 0;
 		reference_lists_info.RefPicList0[0] = static_cast<uint8_t>(*ref_slot);
+
+		if (st_rps.used_by_curr_pic_s0_flag == 0)
+		{
+			// ref_slot is too old → encode an IDR instead
+			ref_slot.reset();
+			poc_history.clear();
+		}
```

Comment:
This happens after `video_cmd_buf.beginVideoCodingKHR` has been called, which already uses ref_slot. I don't think this will work.